core/src/main/java/ch/unibas/dmi/dbis/polyphenydb/statistic/StatisticsManager.java

```diff
@@ -0,0 +1,498 @@
+        // To ensure only one instance is created
+        if ( instance == null ) {
+            instance = new StatisticsManager();
+        }
+        return instance;
+    }
+
+
+    private void insert( String schemaTableColumn, T val ) {
+        String[] splits = QueryColumn.getSplitColumn( schemaTableColumn );
+        insert( splits[0], splits[1], splits[2], val );
+    }
+
+
+    /**
+     * adds data for a column to the store
+     *
+     * @param schema schema name
+     * @param qualifiedTable table name
+     * @param qualifiedColumn column name
+     */
+    private void insert( String schema, String qualifiedTable, String qualifiedColumn, T val ) {
+        PolySqlType type = this.sqlQueryInterface.getColumnType( qualifiedColumn );
+        insert( schema, qualifiedTable, qualifiedColumn, type, val );
+    }
+
+
+    private void insert( String schema, String qualifiedTable, String qualifiedColumn, PolySqlType type, T val ) {
+        if ( !this.statisticSchemaMap.containsKey( qualifiedColumn ) ) {
+            addColumn( qualifiedColumn, type );
+        }
+        getColumn( schema, qualifiedTable, qualifiedColumn ).insert( val );
+    }
+
+
+    /**
+     * Gets the specific statisticColumn if it exists in the tracked columns
+     * else null
+     *
+     * @return the statisticColumn which matches the criteria
+     */
+    private StatisticColumn<T> getColumn( String schema, String table, String column ) {
+        if ( this.statisticSchemaMap.containsKey( schema ) ) {
+            if ( this.statisticSchemaMap.get( schema ).containsKey( table ) ) {
+                if ( this.statisticSchemaMap.get( schema ).get( table ).containsKey( column ) ) {
+                    return this.statisticSchemaMap.get( schema ).get( table ).get( column );
+                }
+            }
+        }
+        return null;
+    }
+
+
+    /**
+     * Adds a new column to the tracked columns and sorts it correctly
+     *
+     * @param qualifiedColumn column name
+     * @param type the type of the new column
+     */
+    private void addColumn( String qualifiedColumn, PolySqlType type ) {
+        String[] splits = QueryColumn.getSplitColumn( qualifiedColumn );
+        if ( type.isNumericalType() ) {
+            put( splits[0], splits[1], splits[2], new NumericalStatisticColumn<>( splits, type ) );
+        } else if ( type.isCharType() ) {
+            put( splits[0], splits[1], splits[2], new AlphabeticStatisticColumn<>( splits, type ) );
+        }
+    }
+
+
+    public void addAll( String schema, String table, String column, PolySqlType type, List<T> vals ) {
+        vals.forEach( val -> {
+            insert( schema, table, column, type, val );
+        } );
+    }
+
+
+    /**
+     * Reset all statistics and reevaluate them
+     */
+    private void reevaluateStore() {
+        if ( this.sqlQueryInterface == null ) {
+            return;
+        }
+        log.warn( "Resetting StatisticStore." );
```

Comment:
Is `warn` the proper log level here? Or could it be lower? (`info` or even `debug`)

(Also for the second log statement in this method)